### ISSUE TITLE
[BUG] [Linter] djlint exclude option does not work as expected

### ISSUE THREAD

**Issue by antoineauger** [djlint/djLint#1028]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info

- OS:  Debian GNU/Linux 12
- Python Version:  Python `3.12.7`
- djLint Version: `djlint, version 1.36.1`
- template language: django (`.html.j2` extension)

## Issue

None of the excluded paths is transformed to an absolute path. Therefore, the check performed in https://github.com/djlint/djLint/blob/master/djlint/src.py#L47 only searches if some of the excluded patterns are contained in the current path (instead of being equal).

If `--exclude=build` is passed, all paths containing `*build*` will be ignored, even if the string only partially match a parent directory (for instance, `/builds/template.html.j2`, `/build-my-template/template.html.j2`, etc. will always be ignored no matter where `djlint` is ran).

### Expected behavior

If `--exclude=build` is passed, the final regex for excluding paths should be `^{SRC}/build/.*$`

## How To Reproduce

See comment below: https://github.com/djlint/djLint/issues/1028#issuecomment-2493530527

<details>
  <summary>Old reproduce steps</summary>
  
Tests performed on a CI runner:

```shell
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

$ poetry run python --version
Python 3.12.7

$ echo $0
/usr/bin/bash

$ poetry run djlint ./ --check
No files to check! 😢

$ poetry run djlint ${CI_PROJECT_DIR}/ --check
No files to check! 😢

$ poetry run djlint ./**/templates/*.html.j2 --check
Checking 6/6 files ━━━━━━━━━━ 00:00    
0 files would be updated.
```

Tests performed locally:

```sh
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

$ poetry run python --version
Python 3.12.7

$ echo $0
zsh

$ poetry run djlint ./ --check
Checking 6/6 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 00:00    
0 files would be updated.

$ poetry run djlint ./**/templates/*.html.j2 --check
Checking 6/6 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 00:00    
0 files would be updated.
```
</details>

## Contents of djlint.toml/.djlintrc/pyproject.toml [tool.djlint]

```toml
[tool.djlint]
ignore = "H005,H006,H016,H017,H021,H030,H031"
extension = "html.j2"
extend_exclude = "frontend,htmlcov"
indent = 2
```

/cc @nejch

**Comment by antoineauger** [comment]
Note: I wanted to debug this a bit more locally/submit a PR but it seems that the standard output is redirected and I cannot manage to display info with `print()` 😅

**Comment by nemovc** [comment]
I've had this issue and have managed to track it down to the default `exclude` regex causing it to reject filepaths which contain 'build'... which is a problem, when e.g., gitlab CI runs in `/builds/<group>/<repo-slug>`.  Are you able to `pwd` inside the CI runner and confirm whether it matches any of the strings listed in https://github.com/djlint/djLint/blob/master/djlint/settings.py#L478 

If so, this is also related to #790

**Comment by antoineauger** [comment]
@nemovc I just made some additional tests and I think you're right 😮 

Given the following tree structure:

```sh
/builds/core/api
   top_level_template.html.j2
   subdir1/
      templates/
            inner_template.html.j2
```

I ran the following tests in CI:

```sh
$ pwd
/builds/core/api

$ poetry run djlint --check . --exclude=build
No files to check! 😢

$ poetry run djlint --check . --exclude=builds
No files to check! 😢

$ poetry run djlint --check . --exclude=venv
Checking 2/2 files ━━━━━━━━━━ 00:00    
0 files would be updated.

$ poetry run djlint --check . --exclude=templates
Checking 1/1 files ━━━━━━━━━━ 00:00    
0 files would be updated.

$ poetry run djlint --check . --exclude=temp
Checking 1/1 files ━━━━━━━━━━ 00:00    
0 files would be updated.
```

I could not see the problem first since the `default_exclude` only contains `build` and GitLab works inside a `builds` directory, so not the same directory name.

The root cause of all of this is that none of the excluded paths is transformed to an absolute path. 
Therefore, the check performed in https://github.com/djlint/djLint/blob/master/djlint/src.py#L47 only searches if some of the excluded patterns are _contained_ in the current path (instead of being equal).

If `--exclude=build` is passed, the final regex for excluding paths should be `^{SRC}/build/.*$`

I'll rename this issue and adjust the description. I think this is the root cause of:
- https://github.com/djlint/djLint/issues/790
- https://github.com/djlint/djLint/issues/800
- https://github.com/djlint/djLint/issues/832

**Comment by antoineauger** [comment]
@monosans Could you reopen this issue after the revert?

Thanks.

**Comment by Roang-zero1** [comment]
Another dimension for me is that the [extend_exclude](https://github.com/djlint/djLint/blob/8f92f2ecaf327dc545b14d07888630eaea3dfb0d/src/djlint/settings.py#L521) is split along `,`, but the [exclude](https://github.com/djlint/djLint/blob/8f92f2ecaf327dc545b14d07888630eaea3dfb0d/src/djlint/settings.py#L511-L513) is not.  
So I can't override the setting with the [examples provided in the docs](https://www.djlint.com/docs/configuration/#exclude).

Therefore I had to work around this like this:
```toml
# Need to be at leas one entry or it breaks
exclude = ".venv"
# Need to set the rest here, otherwise it won't work from exclude
extend_exclude = "__pypackages__,_build,.bzr,.direnv,.eggs,.git,.git-rewrite,.hg,.ipynb_checkpoints,.mypy_cache,.nox,.pants.d,.pytest_cache,.pytype,.ruff_cache,.svn,.tox,.venv,.vscode,buck-out,dist,node_modules,venv,.tools"
```

**Comment by monosans** [comment]
> Another dimension for me is that the [extend_exclude](https://github.com/djlint/djLint/blob/8f92f2ecaf327dc545b14d07888630eaea3dfb0d/src/djlint/settings.py#L521) is split along `,`, but the [exclude](https://github.com/djlint/djLint/blob/8f92f2ecaf327dc545b14d07888630eaea3dfb0d/src/djlint/settings.py#L511-L513) is not.

Thank you for your observation. It was also fixed in v1.38.1.
